cmd/eksctl-anywhere/cmd/checkimages.go

```diff
@@ -0,0 +1,92 @@
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net"
+	"strings"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	"github.com/spf13/viper"
+
+	"github.com/aws/eks-anywhere/pkg/cluster"
+	"github.com/aws/eks-anywhere/pkg/constants"
+	"github.com/aws/eks-anywhere/pkg/executables"
+	"github.com/aws/eks-anywhere/pkg/version"
+)
+
+type checkImagesOptions struct {
+	fileName string
+}
+
+var cio = &checkImagesOptions{}
+
+func init() {
+	rootCmd.AddCommand(checkImagesCommand)
+	checkImagesCommand.Flags().StringVarP(&cio.fileName, "filename", "f", "", "Filename that contains EKS-A cluster configuration")
+	err := checkImagesCommand.MarkFlagRequired("filename")
+	if err != nil {
+		log.Fatalf("Error marking filename flag as required: %v", err)
+	}
+}
+
+var checkImagesCommand = &cobra.Command{
+	Use:   "check-images",
+	Short: "Check images used by EKS Anywhere do exist in the target registry",
+	Long:  "This command is used to check images used by EKS-Anywhere for cluster provisioning do exist in the target registry",
+	PreRunE: func(cmd *cobra.Command, args []string) error {
+		cmd.Flags().VisitAll(func(flag *pflag.Flag) {
+			if err := viper.BindPFlag(flag.Name, flag); err != nil {
+				log.Fatalf("Error initializing flags: %v", err)
+			}
+		})
+		return nil
+	},
+	SilenceUsage: true,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		return checkImages(cmd.Context(), cio.fileName)
+	},
+}
+
+func checkImages(context context.Context, spec string) error {
+	docker := executables.BuildDockerExecutable()
+	images, err := getImages(spec)
+	if err != nil {
+		return err
+	}
+
+	clusterSpec, err := cluster.NewSpecFromClusterConfig(spec, version.Get())
+	if err != nil {
+		return err
+	}
+
+	myRegistry := constants.DefaultRegistry
+
+	if clusterSpec.Cluster.Spec.RegistryMirrorConfiguration != nil {
+		host := clusterSpec.Cluster.Spec.RegistryMirrorConfiguration.Endpoint
+		if len(host) > 0 {
+			port := clusterSpec.Cluster.Spec.RegistryMirrorConfiguration.Port
+			if port == "" {
+				port = constants.DefaultHttpsPort
+			}
+			myRegistry = net.JoinHostPort(host, port)
+		}
+	}
+
+	for _, image := range images {
+		myImageUri := strings.ReplaceAll(image.URI, constants.DefaultRegistry, myRegistry)
+		if image.ImageDigest != "" {
+			fmt.Printf("%s@%s\n", myImageUri, image.ImageDigest)
+		} else {
+			fmt.Printf("%s\n", myImageUri)
+		}
+
+		if err := docker.PullImage(context, myImageUri); err != nil {
```

Comment:
Is there a way we can check if the image exists without pulling it? This would leave the image in the local docker cache.
Maybe with `docker manifest inspect`?